store/root/store.go

```diff
@@ -150,9 +149,28 @@ func (s *Store) GetLatestVersion() (uint64, error) {
 	return lastCommitID.Version, nil
 }
 
-// GetProof delegates the GetProof to the store's underlying SC backend.
-func (s *Store) GetProof(_ string, version uint64, key []byte) (*ics23.CommitmentProof, error) {
-	return s.stateCommitment.GetProof(version, key)
+func (s *Store) Query(storeKey string, version uint64, key []byte, prove bool) (store.QueryResult, error) {
+	val, err := s.stateStore.Get(storeKey, version, key)
+	if err != nil {
+		return store.QueryResult{}, err
+	}
+
+	result := store.QueryResult{
+		Key:     key,
+		Value:   val,
+		Version: version,
+	}
+
+	if prove {
+		proof, err := s.stateCommitment.GetProof(version, key)
```

Comment:
should get proof return the value as well, this would avoid the current two lookups and reduce it to one

Reply:
I like the current implementation, this allows us to get rid of value from the commitment tree, furthermore, we can remove leaves from the tree.

Reply:
since we agreed with removing fast nodes, it's a similar way.

Reply:
fastnode is only for latest state, if you needed historical proofs it would be a single lookup. We can leave it as is, just thinking of the added overhead here